ext/hash/hash_haval.c

```diff
@@ -280,15 +280,16 @@ PHP_HASH_HAVAL_INIT(5,256)
 /* {{{ PHP_HAVALUpdate */
 PHP_HASH_API void PHP_HAVALUpdate(PHP_HAVAL_CTX *context, const unsigned char *input, size_t inputLen)
 {
-	unsigned int i, index, partLen;
+	unsigned int index, partLen;
+	size_t i;
 
 	/* Compute number of bytes mod 128 */
 	index = (unsigned int) ((context->count[0] >> 3) & 0x7F);
 	/* Update number of bits */
 	if ((context->count[0] += ((uint32_t) inputLen << 3)) < ((uint32_t) inputLen << 3)) {
 		context->count[1]++;
 	}
-	context->count[1] += ((uint32_t) inputLen >> 29);
+	context->count[1] += (uint32_t) (inputLen >> 29);
```

Comment:
Does it make sense for master to change those to use ``uint_least64_t`` ? As those _should_ always be defined even on 32 bit where there is not a fixed width 64 bit number.

Reply:
For the counts right? Perhaps, although we have to be careful with endianness and the handling of the counts in the finalize step.

Reply:
Yes, but I didn't think about endianness